generic_expression_patterns_modules/download_recount2_data.R

```diff
@@ -123,9 +119,62 @@ base_dir){
   data_counts_all <- t(data_counts_all)
 
   ## Save counts matrix to file
-  write.table(data_counts_all,
-            paste(local_dir,'/recount2_compendium_data.tsv',sep=""),
-            sep='\t',
-            row.names=TRUE,
-            col.names=NA)
-}
\ No newline at end of file
+  write.table(
+    data_counts_all,
+    paste(local_dir,'/recount2_compendium_data.tsv',sep = ""),
+    sep = '\t',
+    row.names = TRUE,
+    col.names = NA
+  )
+}
+
+# This function downloads all recount2 `rse_gene.Rdata` files and generates a
+# transposed data count file (`t_data_counts.tsv`) based on each `rse_gene.Rdata`.
+function download_all_recount2(metadata_dir, download_dir) {
```

Comment:
This function is based on the original `get_recount2_compendium`. The biggest difference is that it saves the transposed `data_counts` for each project on disk. I have a new function in `process.py` to read the individual `t_data_counts.tsv` on disk and create the final compendium file. The main reason to do this is because the line:
```
data_counts_all <- cbind(data_counts_all, assays(scale_counts(rse_gene))$counts)
``` 
required a huge amount of memory when `data_counts_all` gets big.